### ISSUE TITLE
Effect of reduce_zero_label

### ISSUE THREAD

**Issue by iumyx2612** [open-mmlab/mmsegmentation#1714]
I trained Unet on a Polyp segmentation dataset.  
`gt_semantic_seg` returns a 2D array (H, W) like the image below.  
![image](https://user-images.githubusercontent.com/69593462/175867480-02edb0cd-1e84-414c-aa44-16b528ff3aea.png)
Yellow color is background, purple color is a polyp.  

When setting `reduce_zero_label=True`, background has value 255.  
When setting `reduce_zero_label=False`, background has value 0.  

I trained with both settings.  
This is the result of `reduce_zero_label=True`, `num_classes=2`. Left is predicted mask, right is gt_mask
![image](https://user-images.githubusercontent.com/69593462/175868073-1009e595-935f-454d-b464-ed8d439959f8.png)
As you can see, it predict the whole picture (this one is from the training set) as polyp. It always get 0.55 IoU and see no improvement, you can check it in the log here: 
[20220624_165112.log](https://github.com/open-mmlab/mmsegmentation/files/8988959/20220624_165112.log)

And this is the result of `reduce_zero_label=False`, `num_classes=3`.  
![image](https://user-images.githubusercontent.com/69593462/175868583-1de17cc7-c61f-4381-b4a9-e39d598fbaf3.png)
Sorry I didn't save the log for this one. But the neoplastic polyps class has really low IoU compare to the one with `reduce_zero_label=True`.  

Can you give any insight of this behavior, how is loss calculated if I set `reduce_zero_label=True`?
Fyi, I used `CrossEntropyLoss` like `_base_/model/fcn_unet_s5-d16.py`

Here's my dataset class:
```
class PolypDataset(CustomDataset):
    CLASSES = ("Background", "neoplastic polyps", "non-neoplastic polyps")
    PALETTE = [0, 1, 2]
    def __init__(self, **kwargs):
        super(PolypDataset, self).__init__(
            img_suffix=".jpeg",
            seg_map_suffix=".png",
            **kwargs)
        assert self.file_client.exists(self.img_dir)
```

**Comment by MengzhangLI** [comment]
Hi, what is your annotation file content, does it only contains label `0`, `1` and `2`?

When reduce_zero_label is true, when loading annotation it uses `reduce_zero_label` [here](https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/datasets/pipelines/loading.py#L145-L152) and in evaluation it is [here](https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/core/evaluation/metrics.py#L69-L72).

It seems `reduce_zero_label=False`, `num_classes=3` worked but not training very well. So I suggest you check your annotation file (make comparison with DRIVE, STARE, HRF which we provided) and use some strong models like SegFormer and Swin Transformer to test their performance.

**Comment by iumyx2612** [comment]
> Hi, what is your annotation file content, does it only contains label `0`, `1` and `2`?
> 
> When reduce_zero_label is true, when loading annotation it uses `reduce_zero_label` [here](https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/datasets/pipelines/loading.py#L145-L152) and in evaluation it is [here](https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/core/evaluation/metrics.py#L69-L72).
> 
> It seems `reduce_zero_label=False`, `num_classes=3` worked but not training very well. So I suggest you check your annotation file (make comparison with DRIVE, STARE, HRF which we provided) and use some strong models like SegFormer and Swin Transformer to test their performance.

Hi~ I just realize when setting `reduce_zero_label=True`, the background isn't calculated in loss, which leads to the behavior I mentioned above. I'm trying CNNs model from weak to strong and I'll add Transformer family into my model pool. And yes, my annotation mask only contains `0`, `1` and `2`
